lib/daru/dataframe.rb

```diff
@@ -2302,30 +2301,25 @@ def insert_or_modify_vector name, vector
 
           v = Daru::Vector.new(vector, name: set_name(name), index: @index)
         end
-
         assign_or_add_vector name, v
       end
     end
 
     def assign_or_add_vector name, v
-      #FIXME: fix this jugaad. need to make changes in Indexing itself.
-      begin
-        pos = @vectors[name]
-      rescue IndexError
-        pos = name
-      end
+      pos = @vectors[*name]
 
-      if !pos.kind_of?(Daru::Index) and pos == name and
-        (@vectors.include?(name) or (pos.is_a?(Integer) and pos < @data.size))
-        @data[pos] = v
-      elsif pos.kind_of?(Daru::Index)
-        pos.each do |p|
-          @data[@vectors[p]] = v
-        end
+      # Update an existing vector based on its index
+      if pos.is_a? Daru::Index
+        # If there are multiple vectors associated with this name
+        pos.each { |p| @data[@vectors[p]] = v }
       else
-        @vectors = @vectors | [name] if !@vectors.include?(name)
-        @data[@vectors[name]] = v
+        @data[pos] = v
       end
+
+    rescue IndexError
```

Comment:
Rescuing IndexError here is for the situation when the index which the user specified isn't in the data frame. This suggests that the user intended to add a new vector rather than modify an existing vector. I know it's kind of not a good practice to use error handling in this way but in current scenario nothing else was working and was resulting into making the code more complex. For example initially I had used `@index.include? name` to see if an index exists or not but it was giving false for say `[:a, :foo]` when `[:a, :foo, :bar]` exists and `@index` is a MultiIndex. The same goes for DateTimeIndex.